tests/sample_size/test_sample_size_calculator.py

```diff
@@ -81,83 +82,82 @@ def test_get_single_sample_size_ttest(self, mock_solve_power):
             alternative="two-sided",
         )
 
-    # parameterize get overall sample size function
+    @parameterized.expand(
+        [
+            ("boolean", {"probability": 0.05, "mde": 0.02}, BooleanMetric),
+            ("numeric", {"variance": 500, "mde": 5}, NumericMetric),
+            (
+                "ratio",
+                {
+                    "numerator_mean": 2000,
+                    "numerator_variance": 100000,
+                    "denominator_mean": 200,
+                    "denominator_variance": 2000,
+                    "covariance": 5000,
+                    "mde": 10,
+                },
+                RatioMetric,
+            ),
+        ]
+    )
+    @patch("sample_size.sample_size_calculator.SampleSizeCalculator.get_multiple_sample_size")
     @patch("sample_size.sample_size_calculator.SampleSizeCalculator._get_single_sample_size")
-    def test_get_overall_sample_size_bool(self, mock_get_single_sample_size):
-        test_metric_type = "boolean"
+    def test_get_overall_sample_size_single(
+        self, metric_type, metadata, metric_class, mock_get_single_sample_size, mock_get_multiple_sample_size
+    ):
+        test_metric_type = metric_type
         test_sample_size = 2000
         mock_get_single_sample_size.return_value = test_sample_size
 
-        test_mde = 0.02
-        test_probability = 0.05
-        test_metric_metadata = {"probability": test_probability, "mde": test_mde}
+        test_mde = metadata["mde"]
+        # test_probability = 0.05
+        test_metric_metadata = metadata
         calculator = SampleSizeCalculator()
         calculator.register_metrics([{"metric_type": test_metric_type, "metric_metadata": test_metric_metadata}])
 
         sample_size = calculator.get_sample_size()
 
         self.assertEqual(sample_size, test_sample_size)
         mock_get_single_sample_size.assert_called_once()
-        self.assertIsInstance(mock_get_single_sample_size.call_args[0][0], BooleanMetric)
-        assert_equal(mock_get_single_sample_size.call_args[0][0].probability, test_probability)
-        assert_equal(mock_get_single_sample_size.call_args[0][0].mde, test_mde)
+        mock_get_multiple_sample_size.assert_not_called()
+        self.assertIsInstance(mock_get_single_sample_size.call_args[0][0], metric_class)
+        # assert_equal(mock_get_single_sample_size.call_args[0][0].probability, test_probability)
```

Comment:
I'm fine if we do below instead without expllicitly validating each parameters under each metric 
```python
self.assertEqual(mock_get_single_sample_size.call_args[0][0], calculator.metrics[0])
```